src/HaskellWorks/Data/Aeson.hs

```diff
@@ -18,3 +38,26 @@ readJson :: Read a => String -> String -> Parser a
 readJson t s = case readMaybe s of
   Just a  -> pure a
   Nothing -> fail $ "Could not parse " <> t
+
+-- | Render optional fields as missing in JSON output.
+(.?=) :: (KeyValue p, ToJSON v, Monoid p) => Key -> Maybe v -> p
+(.?=) k mv = case mv of
+  Just v -> k .= v
+  Nothing -> mempty
+
+-- | Render optional fields as having null value in JSON output.
```

Comment:
This isn't exactly accurate anymore. This should instead talk about its precedence (and use with lenses)